File: frontend/src/types/hashtags.ts
export interface Hashtag {
  id: string;
  name: string;
  posts_count: number;
  followers_count: number;
  description: string | null;
  created_at: string;
  is_following: boolean;
}

export interface HashtagDetail {
  id: string;
  name: string;
  posts_count: number;
  followers_count: number;
  description: string | null;
  is_following: boolean;
  created_at: string;
}

export interface TrendingHashtag {
  id: string;
  name: string;
  posts_count: number;
  followers_count: number;
  description: string | null;
  is_following: boolean;
}

export interface HashtagPost {
  id: string;
  content: string | null;
  user_id: string;
  username: string | null;
  user_avatar: string | null;
  post_type: string;
  image_urls: string | null;
  likes_count: number;
  comments_count: number;
  created_at: string;
}

export interface TrendingHashtagsResponse {
  hashtags: TrendingHashtag[];
}

export interface HashtagListResponse {
  hashtags: Hashtag[];
}

export interface HashtagPostsResponse {
  posts: HashtagPost[];
  total_count: number;
}

export interface FollowResponse {
  message: string;
  hashtag_id: string;
  is_following: boolean;
}

File: backend/app/schemas/hashtags.py
from pydantic import BaseModel
from typing import Optional
from uuid import UUID
from datetime import datetime


class HashtagResponse(BaseModel):
    id: UUID
    name: str
    posts_count: int
    followers_count: int
    description: Optional[str] = None
    created_at: datetime
    is_following: bool = False

    model_config = {"from_attributes": True}


class HashtagCreate(BaseModel):
    name: str
    description: Optional[str] = None


class HashtagListResponse(BaseModel):
    hashtags: list[HashtagResponse]


class HashtagDetailResponse(BaseModel):
    id: UUID
    name: str
    posts_count: int
    followers_count: int
    description: Optional[str] = None
    is_following: bool
    created_at: datetime


class TrendingHashtagResponse(BaseModel):
    id: UUID
    name: str
    posts_count: int
    followers_count: int
    description: Optional[str] = None
    is_following: bool = False

    model_config = {"from_attributes": True}


class TrendingHashtagsListResponse(BaseModel):
    hashtags: list[TrendingHashtagResponse]


class HashtagPostResponse(BaseModel):
    id: UUID
    content: Optional[str] = None
    user_id: UUID
    username: Optional[str] = None
    user_avatar: Optional[str] = None
    post_type: str
    image_urls: Optional[str] = None
    likes_count: int
    comments_count: int
    created_at: datetime


class HashtagPostsResponse(BaseModel):
    posts: list[HashtagPostResponse]
    total_count: int
